python_ta/contracts/__init__.py

```diff
@@ -221,6 +225,12 @@ def new_setattr(self: klass, name: str, value: Any) -> None:
                     else:
                         super(klass, self).__delattr__(name)
                     raise AssertionError(str(e)) from None
+        else:
```

Comment:
I think this `else` is a bit too broad: if `caller_self is self` it doesn't need to be added to `__mutated_instances__`, as the RIs should be checked on `self` at the end of the calling method.

Also, it's worth adding a comment to explain what's going on (parallel to the comment in the if branch).

Reply:
Sounds good! Thanks for the keen eye